modules/schema/src/main/java/org/apache/ignite/internal/schema/registry/SchemaRegistryImpl.java

```diff
@@ -86,7 +87,11 @@ public SchemaDescriptor schema(int ver) {
             return desc;
         }
 
-        desc = history.apply(ver);
+        CompletableFuture<SchemaDescriptor> descFut = history.apply(ver);
+
+        if (descFut != null) {
+            desc = descFut.join();
```

Comment:
oh no, we MUST NOT wait on future in our internals

Reply:
yep, this is not mine change, i just move it here ..

Reply:
I will fill the issue after this code will be merged, i can`t create the issue for non merged code.